src/sentry/profiles/flamegraph.py

```
@@ -257,3 +263,278 @@ def get_chunks_from_spans_metadata(
+                Column("start_timestamp"),
+                Column("end_timestamp"),
+            ],
+            where=[
+                project_condition,
+                start_condition,
+                end_condition,
+                profilers_condition,
+            ],
+            # Order by here follows that of the underlying table
+            # as a performance optimization
+            orderby=[
+                OrderBy(Column("project_id"), Direction.DESC),
+                OrderBy(Column("profiler_id"), Direction.DESC),
+                OrderBy(Column("start_timestamp"), Direction.DESC),
+            ],
+            limit=Limit(max_chunks),
+        )
+
+        result = self._query_chunks_for_profilers(query)
+
+        profiler_metas_by_id = {
+            (profiler_meta.project_id, profiler_meta.profiler_id): profiler_meta
+            for profiler_meta in profiler_metas
+        }
+
+        continuous_profile_candidates: list[ContinuousProfileCandidate] = []
+
+        for row in result["data"]:
+            profiler_meta = profiler_metas_by_id[(row["project_id"], row["profiler_id"])]
+            start = datetime.fromisoformat(row["start"]).timestamp()
+            end = datetime.fromisoformat(row["end"]).timestamp()
+
+            continuous_profile_candidates.append(
+                {
+                    "project_id": profiler_meta.project_id,
+                    "profiler_id": profiler_meta.profiler_id,
+                    "chunk_id": row["chunk_id"],
+                    "start": str(int(max(start, profiler_meta.start) * 1.0e9)),
+                    "end": str(int(min(end, profiler_meta.end) * 1.0e9)),
+                }
+            )
+
+        return continuous_profile_candidates
+
+    def _query_chunks_for_profilers(self, query: Query) -> Mapping[str, Any]:
+        """This function is split out for mocking as we cannot write to the
+        profile chunks dataset in tests today"""
+
+        assert self.snuba_params.organization
```

Should we raise an error here? Otherwise an assertion error might bubble up into unexpected places